packages/core/src/index.ts

```diff
@@ -103,18 +106,16 @@ export class Signal<T = any> {
 		let prevOldDeps = oldDeps;
 		currentSignal = this;
 		oldDeps = this._deps;
-		this._deps = new Set();
+		this._deps = new Map();
 
 		return (shouldUnmark: boolean, shouldCleanup: boolean) => {
-			if (shouldUnmark) this._subs.forEach(unmark);
-
 			// Any leftover dependencies here are not needed anymore
 			if (shouldCleanup) {
 				// Unsubscribe from dependencies that were not accessed:
-				oldDeps.forEach(dep => unsubscribe(this, dep));
+				oldDeps.forEach((_, dep) => unsubscribe(this, dep));
 			} else {
-				// Re-subscribe to dependencies that were not accessed:
```

Comment:
This seems to accidentally reverting https://github.com/preactjs/signals/pull/129/files#diff-9a4ceebe7c6f86856371906c3f061d3b56b7457022b05179884a113e7ced67e8R116

Reply:
Good catch!